access/resource_table_acl.go

```diff
@@ -0,0 +1,303 @@
+package access
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/databrickslabs/databricks-terraform/common"
+	"github.com/databrickslabs/databricks-terraform/compute"
+	"github.com/databrickslabs/databricks-terraform/internal"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+// https://docs.databricks.com/security/access-control/table-acls/object-privileges.html#operations-and-privileges
+
+// TableACL defines table access control
+type TableACL struct {
+	Table     string   `json:"table,omitempty"`
+	View      string   `json:"view,omitempty"`
+	Database  string   `json:"database,omitempty"`
+	Principal string   `json:"principal"`
+	Grants    []string `json:"grants,omitempty" tf:"slice_set"`
+	Denies    []string `json:"denies,omitempty" tf:"slice_set"`
+	Owner     string   `json:"owner,omitempty"`
+	ClusterID string   `json:"cluster_id,omitempty"`
+	// SanityCheck bool     `json:"sanity_check,omitempty"`
+}
+
+func (ta *TableACL) actualDatabase() string {
+	if ta.Database == "" {
```

Comment:
Allowing database to not be set and defaulting to `default` seems like handing users a foot-gun. I think forcing them to explicitly specify the Database is a good idea.

Reply:
take a look at `TypeAndKey` & `loadTableACL ` methods for clarification